pygmt/xarray_backend.py

```diff
@@ -0,0 +1,120 @@
+"""
+An xarray backend for reading raster grid/image files using the 'gmt' engine.
+"""
+
+import os
+from pathlib import Path
+from typing import Literal
+
+import xarray as xr
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import build_arg_list
+from pygmt.src.which import which
+from xarray.backends import BackendEntrypoint
+
+
+class GMTBackendEntrypoint(BackendEntrypoint):
+    """
+    Xarray backend to read raster grid/image files using 'gmt' engine.
+
+    Internally, GMT uses the NetCDF C library to read NetCDF files, and GDAL for GeoTIFF
+    files.
+
+    When using :py:func:`xarray.open_dataarray` or :py:func:`xarray.load_dataarray` with
+    ``engine='gmt'``, pass the ``decode_kind`` parameter that can be either:
+
+    - ``grid`` - for reading single-band raster grids
+    - ``image`` - for reading multi-band raster images
+
+    Examples
+    --------
+    Read a single-band NetCDF file using ``decode_kind="grid"``
+
+    >>> import pygmt
+    >>> import xarray as xr
+    >>>
+    >>> da_grid: xr.Dataset = xr.open_dataarray(
+    ...     "@static_earth_relief.nc", engine="gmt", decode_kind="grid"
+    ... )
```

Comment:
This type hint is not necessary:
```suggestion
    >>> da_grid = xr.open_dataarray(
    ...     "@static_earth_relief.nc", engine="gmt", decode_kind="grid"
    ... )
```